src/endpoints/workerskv/write_bulk.rs

```diff
@@ -23,6 +24,26 @@ impl<'a> Endpoint<(), (), Vec<KeyValuePair>> for WriteBulk<'a> {
     fn body(&self) -> Option<Vec<KeyValuePair>> {
         Some(self.bulk_key_value_pairs.clone())
     }
+    fn validate(&self)  -> Result<(), ApiFailure> {
+        if let Some(body) = self.body() {
+            // this matches the serialization in HttpApiClient
+            let len = serde_json::to_string(&body).unwrap().len();
```

Comment:
makes sense, its worth calling that I originally took this line from the ApiClient:
https://github.com/cloudflare/cloudflare-rs/blob/master/src/framework/mod.rs#L93

I am happy to fix it there as well. For now I have just corrected my usage of unwrap. 

This actually highlights another issue with this approach... the serialization in validate must stay in sync with the impl in the ApiClient or we risk not calculating the length incorrectly. This could be addressed by pulling by encapsulating the serialization into a method on endpoint and calling the same method in both places. Happy to take a stab at that... that also may be needed to support https://github.com/cloudflare/cloudflare-rs/issues/26

Reply:
nice link btw, that is helpful